<?xml version="1.0"?>

<launch>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find frontier_exploration)/config/costmap.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find frontier_exploration)/config/costmap.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find frontier_exploration)/config/base_local_planner.yaml" command="load"/>

    <param name="local_costmap/rolling_window" value="true"/>
    <param name="local_costmap/width" value="10.0"/>
    <param name="local_costmap/height" value="10.0"/>

    <param name="base_global_planner" value="navfn/NavfnROS"/>
    <param name="controller_frequency" value="10"/>
  </node> 

  <node pkg="gmapping" type="slam_gmapping" respawn="true" name="slam_gmapping" output="screen">
    <param name="base_frame" value="base_link_robot"/>
    <param name="map_frame" value="map"/>
    <param name="odom_frame" value="odom"/>

    <!-- map size parameters-->
    <param name="xmin" value="-25"/>
    <param name="ymin" value="-15"/>
    <param name="xmax" value="25"/>
    <param name="ymax" value="15"/>

    <remap from="/map" to="/slam_map" />
    <remap from="/map_metadata" to="/slam_map_metadata" />
    <remap from="/scan" to="/front_laser/scan" />
  </node> 

</launch>